src/utils.cpp

```diff
@@ -285,6 +286,94 @@ std::string get_pid_exe(pid_t pid)
   return get_pid_exe(std::to_string(pid));
 }
 
+std::string get_proc_maps(const std::string &pid)
+{
+  std::error_code ec;
+  std_filesystem::path proc_path{ "/proc" };
+  proc_path /= pid;
+  proc_path /= "maps";
+
+  if (!std_filesystem::exists(proc_path, ec))
+    return "";
+
+  return proc_path.string();
+}
+
+std::string get_proc_maps(pid_t pid)
+{
+  return get_proc_maps(std::to_string(pid));
+}
+
+std::unordered_set<std::string> get_mapped_file_paths_for_pid(pid_t pid)
+{
+  std::unordered_set<std::string> paths;
+  std::string maps_path = get_proc_maps(pid);
+  if (maps_path.empty())
+  {
+    LOG(WARNING) << "Maps path is empty";
+    return paths;
+  }
+
+  std::fstream fs(maps_path, std::ios_base::in);
+  if (!fs.is_open())
+  {
+    LOG(WARNING) << "Unable to open procfs mapfile: " << maps_path;
+    return paths;
+  }
+
+  std::string line;
+  // Example mapping:
+  // 7fc8ee4fa000-7fc8ee4fb000 r--p 00000000 00:1f 27168296 /usr/libc.so.6
+  while (std::getline(fs, line))
+  {
+    char buf[PATH_MAX + 1];
+    buf[0] = '\0';
+    auto res = std::sscanf(line.c_str(), "%*s %*s %*x %*s %*u %[^\n]", buf);
+    // skip [heap], [vdso], and non file paths etc...
+    if (res == 1 && buf[0] == '/')
+    {
+      std::string name = buf;
+      if (name.find("(deleted)") == std::string::npos)
+        paths.insert(std::move(name));
+    }
+  }
+
+  return paths;
+}
+
+std::vector<std::string> get_file_paths_for_pid(pid_t pid)
```

Comment:
the method name *get_file_paths_for_pid* seems misleading, maybe add a *mapped* in the name?